compiler/rustc_hir_analysis/src/check/entry.rs

```diff
@@ -190,6 +168,23 @@ fn check_main_fn_ty(tcx: TyCtxt<'_>, main_def_id: DefId) {
         main_def_id,
         expected_sig,
     );
```

Comment:
Can you make `check_function_signature` (or better yet, `Result<(), ErrorGuaranteed>`, and then do an early-return if it had an error? We should only emit one error, ideally.